pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/JdbiQuestionCached.java

```diff
@@ -21,19 +21,40 @@
 
 public class JdbiQuestionCached extends SQLObjectWrapper<JdbiQuestion> implements JdbiQuestion {
     private static final Logger LOG = LoggerFactory.getLogger(JdbiQuestionCached.class);
-    private static RLocalCachedMap<String, List<QuestionDto>> questionKeyToQuestionDtosCache;
+    private static RLocalCachedMap<Long, QuestionDto> questionIdToDtoCache;
+    private static RLocalCachedMap<Long, List<String>> questionIdToTextSuggestionsCache;
+    private static RLocalCachedMap<Long, Long> compositeChildIdToParentIdCache;
+    private static RLocalCachedMap<Long, List<Long>> compositeParentIdToChildIdsCache;
+    private static RLocalCachedMap<String, Long> stableIdInstanceGuidToQuestionIdCache;
 
     public JdbiQuestionCached(Handle handle) {
         super(handle, JdbiQuestion.class);
         initializeCaching();
     }
 
     private void initializeCaching() {
-        if (questionKeyToQuestionDtosCache == null) {
+        if (questionIdToDtoCache == null) {
             synchronized (getClass()) {
-                if (questionKeyToQuestionDtosCache == null) {
-                    questionKeyToQuestionDtosCache = CacheService.getInstance().getOrCreateLocalCache("questionKeyToQuestionDtoCache",
-                            10000);
+                CacheService cacheService = CacheService.getInstance();
+                if (questionIdToDtoCache == null) {
+                    questionIdToDtoCache = cacheService
+                            .getOrCreateLocalCache("questionIdToDtoCache", 1000);
+                }
+                if (questionIdToTextSuggestionsCache == null) {
+                    questionIdToTextSuggestionsCache = cacheService
+                            .getOrCreateLocalCache("questionIdToTextSuggestions", 100);
+                }
+                if (compositeChildIdToParentIdCache == null) {
+                    compositeChildIdToParentIdCache = cacheService
+                            .getOrCreateLocalCache("compositeChildIdToParentIdCache", 1000);
+                }
+                if (compositeParentIdToChildIdsCache == null) {
+                    compositeParentIdToChildIdsCache = cacheService
+                            .getOrCreateLocalCache("compositeParentIdToChildIdsCache", 1000);
+                }
+                if (stableIdInstanceGuidToQuestionIdCache == null) {
+                    stableIdInstanceGuidToQuestionIdCache = cacheService
+                            .getOrCreateLocalCache("stableIdInstanceGuidToQuestionIdCache", 10_000);
```

Comment:
I tried to estimate sensible cache sizes here. Please advice if you have better insights here about size or caching approach.